### PR TITLE
add any gap block init full csc

### PR DESCRIPTION
**Pull request by GalaxySciTech**
add any gap block init full csc

### CODE REVIEW

**Comment by wanwiset25** [comment]
Would be nice to add some guide (in readme) on how to get these numbers from the network. Or maybe even a quick script.

"subnet": {
    "gsbn":1
  },
    "v2esbn": 18666757
  }

File: contracts/FullCheckpoint.sol [hunk 1]
```diff
@@ -47,59 +47,62 @@ contract FullCheckpoint {
...
-                (uint256(block1.roundNumber) << 64) |
+        bytes32 gapHeaderHash = keccak256(gapBlockHeader);
+
+        HeaderReader.ValidationParams memory gapBlock = HeaderReader
+            .getValidationParams(gapBlockHeader);
+        require(gapBlock.number == gsbn, "Invalid Init Block");
+
+        (, address[] memory next) = HeaderReader.getEpoch(gapBlockHeader);
+
+        if (gsbn == 1) {
```
Comment (wgr523) [inline contracts/FullCheckpoint.sol:77]: can you add a comment here to explain what `gsbn==1` means?

File: contracts/FullCheckpoint.sol [hunk 2]
```diff
@@ -47,59 +47,62 @@ contract FullCheckpoint {
...
+            });
+            currentValidators = validators[1];
+        } else {
+            require(next.length > 0, "No Gap Validator Empty");
+            validators[gapBlock.number] = Validators({
+                set: next,
+                threshold: int256((initialValidatorSet.length * certThreshold))
+            });
+            currentValidators = Validators({
+                set: initialValidatorSet,
```
Comment (wgr523) [inline contracts/FullCheckpoint.sol:90]: should be: `set: next` ?

Comment (GalaxySciTech) [reply contracts/FullCheckpoint.sol:90]: snot

**Review by wgr523 — APPROVED** [review]

**Comment by GalaxySciTech** [comment]
> Would be nice to add some guide (in readme) on how to get these numbers from the network. Or maybe even a quick script.
> 
> "subnet": { "gsbn":1 }, "v2esbn": 18666757 }

already finish guide doc in readme

**Pull request merged**